lib/puppet/functions/simplib/validate_port.rb

```diff
@@ -0,0 +1,39 @@
+# Validates whether or not the passed argument is a valid port
+# (i.e. between `1` - `65535`).
+Puppet::Functions.create_function(:'simplib::validate_port') do
```

Comment:
@lnemsick-simp I would like to keep this function, but I think that it can be changed to a pure Puppet function.

```puppet
$foo = 2

unless $foo =~ String { fail("'foo' should be a String, got '${foo}'") }
```